app/uk/gov/hmrc/openidconnect/userinfo/services/UserInfoService.scala

```diff
@@ -47,36 +45,41 @@ trait LiveUserInfoService extends UserInfoService {
       case None => Future.failed(new UnauthorizedException("Bearer token is required"))
     }
 
-    val promiseNino = Promise[Option[Nino]]()
-    val maybeNino = promiseNino.future
-
     scopes flatMap { scopes =>
-      val scopesForNino = Set("profile", "address", "openid:gov-uk-identifiers")
-      if ((scopesForNino -- scopes).size != scopesForNino.size) {
-        authConnector.fetchNino() onComplete {
-          case Success(nino) => promiseNino success nino
-          case Failure(exception) => throw exception
-        }
-      } else promiseNino success None
+      def getMaybeForScopes[T](maybeScopes: Set[String], allScopes: Set[String], f: => Future[Option[T]]): Future[Option[T]] = {
+        if ((maybeScopes -- allScopes).size != maybeScopes.size) f
```

Comment:
Unfortunately `intersects` gives the intersection of the sets. What I need here is only the difference as I am checking if at least one of registered scopes is also in maybe scopes.
The following code shows the difference between `intersects` and `--` on sets:
```
val maybeScopes = Set("a", "c", "d")
val registeredScopes1 = Set("a", "b")
val registeredScopes2 = Set("a", "c", "b", "f")
val registeredScopes3 = Set("a", "c", "d")
val registeredScopes4 = Set("a", "b", "c", "d")
val registeredScopes5 = Set("e", "f", "g")
   
maybeScopes -- registeredScopes1 // gives Set(c, d)
maybeScopes -- registeredScopes2 // gives Set(d)
maybeScopes -- registeredScopes3 // gives Set()
maybeScopes -- registeredScopes4 // gives Set()
maybeScopes -- registeredScopes5 // gives Set(a, c, d)

maybeScopes intersect registeredScopes1 // gives Set(a)
maybeScopes intersect registeredScopes2 // gives Set(a, c)
maybeScopes intersect registeredScopes3 // gives Set(a, c, d)
maybeScopes intersect registeredScopes4 // gives Set(a, c, d)
maybeScopes intersect registeredScopes5 // gives Set()
```